testreport/tasks.py

```diff
@@ -147,3 +153,35 @@ def update_state(bug):
             get_issue_fields_from_bts(bug.externalId)['status']['name']
         log.debug('Saving bug "{}"'.format(bug.externalId))
         bug.save()
+
+
+@celery.task()
+def parse_xml(xunit_format, launch_id, params,
+              s3_conn=False, s3_key_name=None, file_content=None):
+    if s3_conn:
+        s3_connection = get_s3_connection()
+        bucket = s3_connection.get_bucket(settings.S3_BUCKET_NAME)
+        report = bucket.get_key(s3_key_name)
+        file_content = report.read()
+
+    log.info('Start parsing xml {}'.format(s3_key_name))
+    try:
+        xml_parser_func(format=xunit_format,
+                        file_content=file_content,
+                        launch_id=launch_id,
+                        params=params)
+        log.info('Xml parsed successful')
+    except Exception as e:
```

Comment:
If we can't parse uploaded file, maybe it's a reason for file deleting from storage?